### PR TITLE
Validate hostname in e2e integration

### PR DESCRIPTION
**Pull request by clyang82**
**What this PR does**:

Limit the server container name. throw error if the server container name exceeds 64 chars.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: integration/e2e/service.go [hunk 1]
```diff
@@ -235,8 +241,17 @@ func (s *ConcreteService) Ready() error {
 	return s.readiness.Ready(s)
 }
 
-func (s *ConcreteService) containerName() string {
-	return NetworkContainerHost(s.usedNetworkName, s.name)
+func (s *ConcreteService) setContainerName() error {
+	s.containerName = NetworkContainerHost(s.usedNetworkName, s.name)
+	// Linux supports up to 64 chars for hostname
+	if len(s.containerName) > 64 {
+		return fmt.Errorf("service container name %s should not exceed 64.", s.containerName)
```
Comment (alvinlin123) [inline integration/e2e/service.go:248]: Nit: for the error message I would put the offending argument within "[]", so something like "service container name [%s] should not exceed 64."
 
`service container name long long longstring should not exceed 64.`  it's not as easy to be parsed by eyeball as `service container name [long long longstring] should not exceed 64.`

Also I think Go's convention is to not end error message with any punctuation according to https://github.com/golang/go/wiki/CodeReviewComments#error-strings?

**Review by alvinlin123 — APPROVED** [review]

**Comment by bboreham** [comment]
Can you explain the background to this PR please?  I guess you hit some problem, but I don't know what the problem was so it's difficult to see whether your fix is a good one.

**Comment by bboreham** [comment]
I took another look, and it seems the problem could have been a really long network name, because the two are concatenated to form the container name via `NetworkContainerHost()`.

I think all the container names are hard-coded in tests, so the place it can go wrong is when we take the value from environment variable `E2E_NETWORK_NAME`.  All that being correct, it seems simpler to me to put a constraint on that value and fail at the beginning of the test, perhaps by writing a `TestMain()`.

File: integration/e2e/service.go (hunk 1 above)
Comment (clyang82) [reply integration/e2e/service.go:248]: Sorry for late response. Thanks for your great comment about the guideline of error message. I will update soon.

**Comment by clyang82** [comment]
> Can you explain the background to this PR please? I guess you hit some problem, but I don't know what the problem was so it's difficult to see whether your fix is a good one.

Here is a bug I have fixed in thanos community - https://github.com/thanos-io/thanos/pull/4516
```
06:22:50 toolsBucketWeb-1: level=error name=toolsBucketWeb-1 ts=2021-08-03T06:22:50.42873554Z caller=runutil.go:101 msg="function failed. Retrying in next tick" err="BaseFetcher: iter bucket: Get \"http://e2e_test_tools_bucket_web_external_prefix_and_route_prefix-minio-8080:8080/toolsBucketWeb_test/?location=\": dial tcp: lookup e2e_test_tools_bucket_web_external_prefix_and_route_prefix-minio-8080: no such host"
```

**Comment by bboreham** [comment]
I agree with the suggestion on that PR https://github.com/thanos-io/thanos/pull/4516#pullrequestreview-720873238 - it seems simpler to validate the parameter to `NewScenario()`.

**Pull request closed without merging**